src/main/java/com/epam/exception/RestResponseEntityExceptionHandler.java

```diff
@@ -0,0 +1,87 @@
+package com.epam.exception;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import javassist.NotFoundException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.hibernate.exception.ConstraintViolationException;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.dao.InvalidDataAccessApiUsageException;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.transaction.TransactionSystemException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import java.io.FileNotFoundException;
+
+@ControllerAdvice
+public class RestResponseEntityExceptionHandler extends ResponseEntityExceptionHandler {
+    private static final Logger LOG = LogManager.getLogger();
+    private static final HttpHeaders headers = new HttpHeaders();
```

Comment:
why static?

Reply:
As written in this article: https://logging.apache.org/log4j/2.x/manual/usage.html, generally, it is better to declare Loggers as static, because:
1. Instantiation of a new Logger is a fairly expensive operation when using the default ContextSelector, ClassLoaderContextSelector.
2. Once a Logger is created it will not be deleted until the LoggerContext it is associated with is deleted. Typically, this will only happen when the application is shut down or un-deployed.
3. There is no behavioral difference between a static and non-static Logger.